crates/fluvio/src/smartmodule.rs

```diff
@@ -0,0 +1,109 @@
+use std::io::Read;
+
+use flate2::read::GzDecoder;
+use fluvio_protocol::ByteBuf;
+use fluvio_sc_schema::smartmodule::SmartModuleSpec;
+
+use crate::{FluvioAdmin, FluvioError};
+
+pub use fluvio_spu_schema::server::smartmodule::SmartModuleInvocation;
+pub use fluvio_spu_schema::server::smartmodule::SmartModuleInvocationWasm;
+pub use fluvio_spu_schema::server::smartmodule::SmartModuleKind;
+pub use fluvio_spu_schema::server::smartmodule::SmartModuleContextData;
+pub use fluvio_smartmodule::dataplane::smartmodule::SmartModuleExtraParams;
+
+#[cfg(feature = "smartengine")]
+pub use fluvio_smartengine::{SmartModuleChainBuilder, SmartModuleConfig, SmartModuleInitialData};
+
+/// Provides an interface for reading SmartModules from a Fluvio cluster
+///
+/// Requires administrator access to the Fluvio cluster.
+///
+/// # Example
+/// ```
+/// use fluvio::{Fluvio, FluvioError};
+/// use fluvio::smartmodule::{FluvioSmartModule, SmartModule};
+///
+/// async fn do_get_smartmodule() -> Result<Option<SmartModule>, FluvioError> {
+///  let fluvio_smartmodule = Fluvio::connect().await?.smartmodule().await;
+///  let smart_module = fluvio_smartmodule.fetch("group/name@version").await?;
+///  Ok(smart_module)
+/// }
+/// ```
+pub struct FluvioSmartModule {
```

Comment:
This is bit ambigious  .  Maybe `SmartModuleAdmin` ? or `SmartModuleManager`

Reply:
done. Renamed to `SmartModuleAdmin`